libs/prelude/Prelude/Algebra.idr

```diff
@@ -286,44 +160,19 @@ class (VerifiedBoundedJoinSemilattice a, VerifiedBoundedMeetSemilattice a, Verif
 ||| + Unity for `<.>`:
 |||     forall a,     a <.> unity     == a
 |||     forall a,     unity <.> a     == a
-||| + InverseM of `<.>`:
-|||     forall a,     a <.> inverseM a == unity
-|||     forall a,     inverseM a <.> a == unity
+||| + InverseM of `<.>`, except for neutral
+|||     forall a /= neutral,  a <.> inverseM a == unity
+|||     forall a /= neutral,  inverseM a <.> a == unity
 ||| + Distributivity of `<.>` and `<->`:
 |||     forall a b c, a <.> (b <+> c) == (a <.> b) <+> (a <.> c)
 |||     forall a b c, (a <+> b) <.> c == (a <.> c) <+> (b <.> c)
-class RingWithUnity a => Field a where
-  inverseM : a -> a
+class (RingWithUnity a, Eq a) => Field a where
+  inverseM : (x : a) -> (notId : x == neutral = False) -> a
```

Comment:
As for the `So` option, wouldn't that, by nature of being a run-time check, make it inherently difficult to use `inverseM` in functions that pass the type checker? I never use `So`, so maybe I'm mistaken about that.

I like the `(Not (x = neutral))` option much better, and I tried it myself. I just encountered a [type class resolution error](https://gist.github.com/BlackBrane/6c277f97cc565d1bf684) when trying to define `VerifiedField` in the straightforward way, and so I proceeded with this definition instead. But maybe you or someone else can resolve it; it certainly seems like it shouldn't really be an issue. 

Also, along the lines of the previous question: Would this be more difficult to use in practice? What does it look like in general to supply a proof of `Not (x = neutral)`?